src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -489,69 +473,84 @@ end
 # Implementation due to Seeger, Matthias, et al. "Auto-differentiating linear algebra."
 function rrule(
     ::typeof(cholesky),
-    A::LinearAlgebra.HermOrSym{<:LinearAlgebra.BlasReal, <:StridedMatrix},
+    A::LinearAlgebra.RealHermSymComplexHerm{<:Real, <:StridedMatrix},
     ::Val{false};
     check::Bool=true,
 )
     C = cholesky(A, Val(false); check=check)
-    function _cholesky_HermOrSym_pullback(ΔC::Tangent)
-        Ā, U = _cholesky_pullback_shared_code(C, ΔC)
-        Ā = BLAS.trsm!('R', 'U', 'C', 'N', one(eltype(Ā)) / 2, U.data, Ā)
+    function cholesky_HermOrSym_pullback(ΔC)
+        Ā = _cholesky_pullback_shared_code(C, unthunk(ΔC))
+        rmul!(Ā, one(eltype(Ā)) / 2)
```

Comment:
What's the reason for not using
```suggestion
        rdiv!(Ā, 2)
```
or
```suggestion
        ldiv!(2, Ā)
```
? That seems more direct and simpler.

Reply:
`rdiv!` performs elementwise divisions, so $n^2$ division operations, whereas `rmul!` by the reciprocal performs $n^2$ elementwise multiplications and a single division. Division is generally more expensive than multiplication, so this is cheaper e.g.

```julia
julia> using BenchmarkTools

julia> foo(x, a) = rmul!(copy(x), inv(a));

julia> bar(x, a) = rdiv!(copy(x), a);

julia> x = randn(100, 100);

julia> @btime foo($x, 2.0);
  4.718 μs (2 allocations: 78.17 KiB)

julia> @btime bar($x, 2.0);
  9.003 μs (2 allocations: 78.17 KiB)
```

Reply:
But isn't that something that should be optimized in base? We want to *divide* by 2, so the natural thing to do would be to use a division operator instead of manually working with `eltype`.

Reply:
BTW interestingly the difference seems to be smaller on my computer:
```julia
julia> @btime foo($x, 2.0);
  4.690 μs (2 allocations: 78.17 KiB)

julia> @btime bar($x, 2.0);
  7.098 μs (2 allocations: 78.17 KiB)

julia> VERSION
v"1.7.3"
```

Reply:
FWIW, `rmul!` used this way (to perform a division) is quite common throughout this codebase. e.g. all throughout https://github.com/JuliaDiff/ChainRules.jl/blob/main/src/rulesets/LinearAlgebra/dense.jl and https://github.com/JuliaDiff/ChainRules.jl/blob/main/src/rulesets/LinearAlgebra/lapack.jl. Base Julia itself uses this strategy: https://github.com/JuliaLang/julia/blob/b4eb88a71f8c2d8343b21d8fdd1ec403073a222c/stdlib/LinearAlgebra/src/dense.jl#L1595

So I don't think it's unreasonable to use it here for the extra performance. That being said, this is not the computational bottleneck.